src/sql/parts/connection/connectionDialog/connectionDialogWidget.ts

```diff
@@ -96,6 +98,22 @@ export class ConnectionDialogWidget extends Modal {
 		super(localize('connection', 'Connection'), TelemetryKeys.Connection, _partService, telemetryService, contextKeyService, { hasSpinner: true, hasErrors: true });
 	}
 
+	public refresh(): void {
+		let filteredProviderTypes = this.providerTypeOptions;
+
+		if (this._newConnectionParams &&  this._newConnectionParams.providers) {
+			let validProviderNames = Object.keys(this.providerNameToDisplayNameMap).filter(x => this.includeProvider(x, this._newConnectionParams));
+			if (validProviderNames) {
```

Comment:
This will always be true since `if ([])` is true. Should it be an empty list comparison instead?